client/my-sites/site-settings/form-general.jsx

```diff
@@ -127,35 +120,68 @@ const FormGeneral = React.createClass( {
 		return this.recordEventOnce.bind( this, key, eventAction );
 	},
 
+	setDirtyField( key ) {
+		const newState = {};
+		const dirtyFields = this.state.dirtyFields || [];
+		if ( dirtyFields.indexOf( key ) === -1 ) {
+			newState.dirtyFields = [ ...dirtyFields, key ];
+		}
+		this.setState( newState );
```

Comment:
Updating state here with no changes will still trigger a render. If the intention isn't specifically to do that then I think we should just return early
```js
const dirtyFields = this.state.dirtyFields || [];
if ( dirtyFields.indexOf( key ) >= 0 ) {
    return;
}
this.setState( { dirtyFields: [ ...dirtyFields, key ] } );
```